util/file_reader_writer.cc

```diff
@@ -453,23 +449,45 @@ class ReadaheadRandomAccessFile : public RandomAccessFile {
 
   virtual Status Read(uint64_t offset, size_t n, Slice* result,
                       char* scratch) const override {
+
     if (n + alignment_ >= readahead_size_) {
-      return file_->Read(offset, n, result, scratch);
-    }
 
-    // On Windows in unbuffered mode this will lead to double buffering
-    // and double locking so we avoid that.
-    // In normal mode Windows caches so much data from disk that we do
-    // not need readahead.
-    if (forward_calls_) {
+      // Must use the aligned buffer on direct IO
```

Comment:
We don't have to do this because if direct IO enabled, we are sure the offset here is already aligned.
https://github.com/facebook/rocksdb/blob/master/util/file_reader_writer.cc#L83